cloudevents/emitter_test.go

```diff
@@ -0,0 +1,132 @@
+package cloudevents
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/google/go-cmp/cmp"
+)
+
+func makeClient(t *testing.T) (c client.Client, p *http.Protocol) {
+	p, err := http.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err = client.New(p)
+	if err != nil {
+		t.Errorf("failed to make client %s", err.Error())
+	}
+	return
+}
+
+func receiveEvents(t *testing.T, ctx context.Context, events chan<- event.Event) {
+	c, _ := makeClient(t)
+	go func() {
+		err := c.StartReceiver(ctx, func(ctx context.Context, event event.Event) error {
+			go func() {
+				events <- event
+			}()
+			return nil
+		})
+		if err != nil {
+			t.Errorf("failed to start receiver %s", err.Error())
+		}
+	}()
+	time.Sleep(1 * time.Second) // let the server start
+}
+
+func TestEmitterDefaults(t *testing.T) {
+	events := make(chan event.Event)
+	ctx, cancel := context.WithCancel(context.TODO())
+
+	// start a cloudevent client that receives events
+	// and sends them to a channel
+	receiveEvents(t, ctx, events)
+
+	emitter := NewEmitter()
+	emitter.Emit("http://localhost:8080")
+
+	// received event
+	got := <-events
+
+	cancel() // stop the client
+
+	if got.Source() != "/boson/fn" {
```

Comment:
What if we used `github.com/stretchr/testify/assert` for prettier assertions?

Reply:
I like what assertion frameworks do in some languages, and have used several throughout the years, but (and I stress this is just a humble opinion), I have come to not like them.  especially in Go. Having watched them steadily descend into madness repeatedly throughout the years, they are something I now avoid.

The standard built in language logic operators are are not only accessible to everyone without the need to learn a new API, but they also age better, and tend to teach the reader of the tests about the language, and the package being tested, better as one reads the tests.  The code is being presented in a context similar to how it will be used in their code, rather than a somewhat contrived context of an _almost_, but not quite, natural language assertion syntax that is often just qirky enough to be confusing.

I am happy to go along with the majority opinion on this one, if others want to use an assertions framework I am game, but I feel compelled to make my jaded curmudgeon of a voice heard.  I am also open to being educated on why they are worth the aforementioned problems!